### PR TITLE
Added silent authentication

### PR DESCRIPTION
**Pull request by kzkalra**
Before creating a pull request, make sure that you have read the contribution file located at

https://github.com/pnp/powerShell/blob/dev/CONTRIBUTING.md

## Type ##
- [ ] Bug Fix
- [X] New Feature
- [ ] Sample

## Related Issues? ##
Fixes #X, partially fixes #Y, mentioned in #Z, etc.

## What is in this Pull Request ? ##
Added documentation on how the users can silently authenticate using credentials in Azure DevOps pipelines.

### Guidance ###
* You can delete this section when you are submitting the pull request.* 
* *Please update this PR information accordingly. We use this as part of our release notes in monthly communications.*
* **Please target your PR to Dev branch. If you do not target the Dev branch we will not accept this PR.**

### CODE REVIEW

File: pages/articles/authentication.md [hunk 1]
```diff
@@ -155,3 +155,44 @@ Permission | Permission type | Id | Type
 Connect-PnPOnline [yourtenant].sharepoint.com -Interactive -ClientId [clientid] -Tenant [yourtenant].onmicrosoft.com -AzureEnvironment [USGovernment|USGovernmentHigh|USGovernmentDoD|Germany|China]
 ```
 The AzureEnvironment parameter only allows one value. Select the correct one that matches your cloud deployment.
+
+## Silent Authentication with Credentials for running in Pipelines
+
+For running `Connect-PnPOnline` with user credentials in Azure DevOps pipeline, you need to make sure that authentication in your Azure AD application is configured to allow public client. 
+
+Public client can be configured from the Azure portal from the Authentication Blade in the application or by setting the `allowPublicClient` property in the application's manifest to true.
```
Comment (gautamdsheth) [inline pages/articles/authentication.md:163]: Can you also please add a screenshot for this ?

File: pages/articles/authentication.md [hunk 2]
```diff
@@ -155,3 +155,44 @@ Permission | Permission type | Id | Type
...
+
+`username` and `password` for service account can be stored as secret pipeline variables and can be referenced in the script to achieve complete automation.
```
Comment (gautamdsheth) [inline pages/articles/authentication.md:165]: Can you also please add a screenshot for this ?

File: pages/articles/authentication.md [hunk 3]
```diff
@@ -155,3 +155,44 @@ Permission | Permission type | Id | Type
...
+- Map the names of the regions in your geography to the format used in the weekly file, following the format of AzureCloud., such as AzureCloud.westus. You can map the names of the regions from the Azure Geography list to the format used in the weekly file.
+- You can find the weekly IP file at the following URL: https://www.microsoft.com/en-us/download/details.aspx?id=56519
+
+For example, if your organization is located in the South East Asia region, you would map it to the format AzureCloud.SouthEastAsia.
+
+### Create a named location in Azure AD conditional access
+
+- Go to Azure AD conditional access
+- Open named location blade, click on `+ IP Ranges Location`
+- Enter the IP ranges for Microsoft Hosted Agents, `Mark as trusted location` should be checked.
```
Comment (gautamdsheth) [inline pages/articles/authentication.md:181]: Can you also please add a screenshot or a gif for this ?

File: pages/articles/authentication.md [hunk 4]
```diff
@@ -155,3 +155,44 @@ Permission | Permission type | Id | Type
...
+
+### Create a conditional access policy
+
+- Go to Azure AD conditional access, click on `+New Policy`.
+- Give a meaningful name, click on Users and Groups -> Include select users and groups, select the user with which you want to run your pipeline.
+- Include all cloud apps.
+- Under conditions -> locations include `any locations` and exclude the recently created named location.
+- Under grant -> choose `grant access`. Only `require multifactor authentication needs to be checked`.
+- Enable the policy and click on Save.
```
Comment (gautamdsheth) [inline pages/articles/authentication.md:190]: Can you also please add a screenshot or gif image for this ?

**Review by gautamdsheth — CHANGES_REQUESTED** [review]
hi @kzkalra , the docs update looks good. Thanks for adding that 😊🙏

Have left some review comments, would just request you to add some screenshots.

File: pages/articles/authentication.md (hunk 2 above)
Comment (kzkalra) [reply pages/articles/authentication.md:166]: Done. Thanks for the review.

File: pages/articles/authentication.md (hunk 3 above)
Comment (kzkalra) [reply pages/articles/authentication.md:183]: Done. Thanks for the review.

File: pages/articles/authentication.md (hunk 1 above)
Comment (kzkalra) [reply pages/articles/authentication.md:163]: Done. Thanks for the review.

Comment (gautamdsheth) [reply pages/articles/authentication.md:163]: @kzkalra  - these images need to be in PnP PowerShell repo.

Can you please add them in `pages > images` and create a new folder there and then reference them in docs:

image

Comment (kzkalra) [reply pages/articles/authentication.md:163]: Thanks for the review @gautamdsheth . I have made the suggested changes.

Comment (kzkalra) [reply pages/articles/authentication.md:163]: @gautamdsheth Requesting for re-review. :)

**Comment by KoenZomers** [comment]
Thanks @kzkalra for your contribution! Just curious here about your scenario, why would you opt for loggin on using an username/password above using a clientid/client certificate. Any particular scenario that would otherwise not work?

**Comment by kzkalra** [comment]
Hi @KoenZomers, thanks for reaching out. There are certain scenarios where delegated permissions are mandatory, so in these cases we go with username/password or 'delegation'. Some of the common scenarios are listed here : [Limitations of app-only](https://learn.microsoft.com/en-us/sharepoint/dev/solution-guidance/security-apponly#what-are-the-limitations-when-using-app-only)

**Comment by gautamdsheth** [comment]
@KoenZomers  - ok with you ? Looks good to me , you can merge it if it looks good :)

**Pull request merged**